_shared/project/db.py

```diff
@@ -0,0 +1,62 @@
+import alembic.command
+import alembic.config
+{% if cookiecutter.get("_directory") == "pyramid-app" %}
+import zope.sqlalchemy
+{% endif %}
+from sqlalchemy import MetaData{% if cookiecutter.get("_directory") == "pyramid-app" %}, create_engine{% endif %}, text
+from sqlalchemy.exc import ProgrammingError
+from sqlalchemy.orm import DeclarativeBase, MappedAsDataclass{% if cookiecutter.get("_directory") == "pyramid-app" %}, Session{% endif %}
+
+
+
+class Base(MappedAsDataclass, DeclarativeBase):
+    metadata = MetaData(
+        naming_convention={
+            "ix": "ix_%(column_0_N_label)s",
+            "uq": "uq_%(table_name)s_%(column_0_N_name)s",
+            "ck": "ck_%(table_name)s_%(constraint_name)s",
+            "fk": "fk_%(table_name)s_%(column_0_N_name)s_%(referred_table_name)s_%(referred_column_0_N_name)s",
+            "pk": "pk_%(table_name)s",
+        }
+    )
+
+
+def stamp(engine):  # pragma: no cover
+    """If the DB isn't already stamped then stamp it with the latest revision."""
+    stamped = False
+
+    with engine.connect() as connection:
+        try:
+            if connection.execute(text("select * from alembic_version")).first():
+                stamped = True
+        except ProgrammingError:
+            pass
+
+    if not stamped:
+        alembic.command.stamp(alembic.config.Config("conf/alembic.ini"), "head")
+{% if cookiecutter.get("_directory") == "pyramid-app" %}
+
+
+def includeme(config):  # pragma: no cover
+    engine = create_engine(config.registry.settings["database_url"])
+
+    def db(request):
+        """Return the SQLAlchemy session for the given request."""
+        session = Session(engine, info={"request": request})
+
+        # Register the session with pyramid_tm/transaction so that they'll
+        # create a new DB transaction for each request and close the session at
+        # the end of the request.
+        zope.sqlalchemy.register(session, transaction_manager=request.tm)
+
+        return session
+
+    if config.registry.settings.get("dev"):
+        # Create the database tables if they don't already exist.
+        Base.metadata.create_all(engine)
+        stamp(engine)
```

Comment:
Create all the DB tables from the models. This is what causes the tables to be created for new models if you just add a new model to your code and then run `make dev`. And we only want that to happen in dev, not prod! In prod we only want to use DB migrations to change the schema. This is the same as we do in other apps